core/src/test/java/org/testcontainers/junit/CustomTraitTest.java

```diff
@@ -0,0 +1,43 @@
+package org.testcontainers.junit;
+
+import com.google.common.util.concurrent.Uninterruptibles;
+import org.apache.commons.io.IOUtils;
+import org.junit.Rule;
+import org.junit.Test;
+import org.rnorth.ducttape.unreliables.Unreliables;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.traits.ExposedPort;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.Socket;
+import java.util.concurrent.TimeUnit;
+
+import static org.rnorth.visibleassertions.VisibleAssertions.assertEquals;
+
+public class CustomTraitTest {
+
+    @Rule
+    public GenericContainer container = new GenericContainer<>()
+            .with((container1, createContainerCmd) -> createContainerCmd.withHostName("some.domain.com"))
```

Comment:
With Trait having a single abstract method I can see this working OK but if we're potentially going to add other lifecycle event hooks we might need to be careful not to break this, right?

Perhaps new lifecycle event hooks would need to be added to Trait as `default` methods to leave configure as a SAM... What do you think?